src/woo_search/search_index/typing.py

```diff
@@ -52,7 +53,10 @@ class PublicationType(TypedDict):
     verkorte_titel: str
     omschrijving: str
     registratiedatum: datetime
+    gepubliceerd_op: datetime
     laatst_gewijzigd_datum: datetime
+    datum_begin_geldigheid: datetime | None
+    datum_einde_geldigheid: datetime | None
```

Comment:
```suggestion
    datum_begin_geldigheid: NotRequired[datetime | None]
    datum_einde_geldigheid: NotRequired[datetime | None]
```
the keys may be entirely absent from the request body

If you specify on the serializer `default=None`, then the keys will be guaranteed present though